### ISSUE TITLE
Allow adding aliases to command using object initializer

### ISSUE THREAD

**Issue by CCRcmcpe** [dotnet/command-line-api#1262]
This is a simple example:
```cs
var rootCommand = new RootCommand
{
    new Command("do-something")
    {
        new Command("abcdefg") // How to add aliases here?
        new Command("qwerty")
    }
};
```
In this case, you have to give up using the object initializer and manually add the commands.
This supposed example demonstrates an easy way to build commands.
```cs
var rootCommand = new RootCommand
{
    new Command("do-something")
    {
        new Command("abcdefg") { Aliases = { "c", "w" } } // In reality, this will raise a compiler error, because Aliases is a IReadOnlyList<string>
        new Command("qwerty")
    }
};
```

**Comment by elgonzo** [comment]
Just an additional note: If the public API of `Command` is being updated/changed to allow such functionality, please also make it consistent with `Option`. Currently, `Option`\`Option<T>` allows specifying aliases through constructor arguments and an `AddAlias` method, whereas a `Command` exposes the same feature only through an `AddAlias` method.

**Comment by CCRcmcpe** [comment]
Agreed. More generally, the whole construction of the command structure should be done in one go by the object initializer.

**Comment by kzu** [comment]
The whole pack, Option/Argument/Command should probably be changed to records so this all comes mostly for free :)

**Comment by DavidVTurley** [comment]
Would also be great if we can use the .AddAlias() in a fluent way.
new Command()
           .AddAlias()
           .AddAlias();
